modules/navier_stokes/src/auxkernels/INSFVSlipVelocityAux.C

```diff
@@ -0,0 +1,143 @@
+  params.addParam<MooseFunctorName>("fd", 0.0, "Fraction dispersed phase.");
+  MooseEnum momentum_component("x=0 y=1 z=2");
+  params.addRequiredParam<MooseEnum>(
+      "momentum_component",
+      momentum_component,
+      "The component of the momentum equation that this kernel applies to.");
+  return params;
+}
+
+INSFVSlipVelocityAux::INSFVSlipVelocityAux(const InputParameters & params)
+  : AuxKernel(params),
+    _dim(_subproblem.mesh().dimension()),
+    _u_var(dynamic_cast<const INSFVVelocityVariable *>(getFieldVar("u", 0))),
+    _v_var(params.isParamValid("v")
+               ? dynamic_cast<const INSFVVelocityVariable *>(getFieldVar("v", 0))
+               : nullptr),
+    _w_var(params.isParamValid("w")
+               ? dynamic_cast<const INSFVVelocityVariable *>(getFieldVar("w", 0))
+               : nullptr),
+    _rho_mixture(getFunctor<ADReal>(NS::density)),
+    _rho_d(getFunctor<ADReal>("rho_d")),
+    _mu_mixture(getFunctor<ADReal>(NS::mu)),
+    _gravity(getParam<RealVectorValue>("gravity")),
+    _force_scale(getParam<Real>("force_value")),
+    _force_function(getFunction("force_function")),
+    _force_postprocessor(getPostprocessorValue("force_postprocessor")),
+    _force_direction(getParam<RealVectorValue>("force_direction")),
+    _linear_friction(getFunctor<ADReal>("linear_coef_name")),
+    _particle_diameter(getFunctor<ADReal>("particle_diameter")),
+    _index(getParam<MooseEnum>("momentum_component"))
+{
+  if (!_u_var)
+    paramError("u", "the u velocity must be an INSFVVelocityVariable.");
+
+  if (_dim >= 2 && !_v_var)
+    paramError("v",
+               "In two or more dimensions, the v velocity must be supplied and it must be an "
+               "INSFVVelocityVariable.");
+
+  if (_dim >= 3 && !_w_var)
+    paramError("w",
+               "In three-dimensions, the w velocity must be supplied and it must be an "
+               "INSFVVelocityVariable.");
+}
+
+Real
+INSFVSlipVelocityAux::computeValue()
+{
+  constexpr Real offset = 1e-15;
+  const auto elem_arg = makeElemArg(_current_elem);
+  const auto state = determineState();
+
+  bool is_transient = _subproblem.isTransient();
+  ADRealVectorValue term_advection;
+  ADRealVectorValue term_transient;
+  ADRealVectorValue term_gravity(_gravity);
+  ADRealVectorValue term_force(_force_scale * _force_postprocessor *
+                               _force_function.value(_t, _current_elem->vertex_average()) *
+                               _force_direction);
+
+  // Adding transient term
+  if (is_transient)
+  {
+    term_transient(0) += _u_var->dot(elem_arg, state);
+    if (_dim > 1)
+      term_transient(1) += _v_var->dot(elem_arg, state);
+    if (_dim > 2)
+      term_transient(2) += _w_var->dot(elem_arg, state);
+  }
+
+  // Adding advection term
+  ADRealVectorValue u_grad = _u_var->gradient(elem_arg, state);
+  term_advection(0) += (*_u_var)(elem_arg, state) * u_grad(0);
```

Comment:
you should save (*_u_var)(elem_arg, state) somewhere to avoid calling it plenty of times

Reply:
same for v and w

Reply:
Thanks. Changed